prefect_dbt/cloud/jobs.py

```diff
@@ -630,3 +638,134 @@ def retry_dbt_cloud_job_run_subset_and_wait_for_completion_flow():
         poll_frequency_seconds=poll_frequency_seconds,
     )
     return run_data
+
+
+class DbtCloudJob(Block, ABC):
+    """
+    placeholder
+    """
+
+    _block_schema_capabilities = [
+        "trigger-job",
+        "wait-for-job-completion",
+        "fetch-job-results",
+    ]
+
+    dbt_cloud_credentials: DbtCloudCredentials
+
+    @property
+    def logger(self):
+        try:
+            return get_run_logger()
+        except MissingContextError:
+            return get_logger()
+
+    @sync_compatible
+    async def trigger(self, job_id, **trigger_job_run_options):
+        """
+        placeholder
+        """
+        triggered_run_data_future = await trigger_dbt_cloud_job_run.submit(
```

Comment:
I don't think we should call tasks inside of block methods because then we're necessitating that these methods be called within a flow. Instead, I think that block methods should be called within tasks.